Google.PowerShell.IntegrationTests/PubSub/PubSub.Subscription.Tests.ps1

```diff
@@ -0,0 +1,163 @@
+        }
+    }
+
+    It "should work with -PushEndpoint" {
+        $r = Get-Random
+        $topicName = "gcp-test-new-subscription-topic-$r"
+        $subscriptionName = "gcp-test-new-subscription-$r"
+        # We created an app and this endpoint is set up for us.
+        $endpoint = "https://gcloud-powershell-testing.appspot.com/_ah/push-handlers/"
+
+        try {
+            New-GcpsTopic -Topic $topicName
+            New-GcpsSubscription -Subscription $subscriptionName `
+                                 -Topic $topicName `
+                                 -PushEndpoint $endpoint
+
+            $subscription = GetSubScription -subscription $subscriptionName -topic $topicName
+            $subscription | Should Not BeNullOrEmpty
+            $subscription.AckDeadlineSeconds | Should Be 10
+            $subscription.PushConfig.PushEndPoint | Should Be $endpoint
+        }
+        finally {
+            gcloud beta pubsub topics delete $topicName --quiet 2>$null
+            gcloud beta pubsub subscriptions delete $subscriptionName --quiet 2>$null
+        }
+    }
+
+    It "should error out for bad subscription name" {
+        $r = Get-Random
+        $topicName = "gcp-test-new-subscription-topic-$r"
+
+        try {
+            New-GcpsTopic -Topic $topicName
+            { New-GcpsSubscription -Topic $topicName -Subscription "!!" -ErrorAction Stop } | Should Throw "Invalid resource name"
+        }
+        finally {
+            gcloud beta pubsub topics delete $topicName --quiet 2>$null
+        }
+    }
+
+    It "should error out for bad ack deadline" {
+        $r = Get-Random
+        $topicName = "gcp-test-new-subscription-topic-$r"
+        $subscriptionName = "gcp-test-new-subscription-$r"
+
+        try {
+            New-GcpsTopic -Topic $topicName
+            # Ack deadline has to be between 10 and 600.
+            { New-GcpsSubscription -Topic $topicName -Subscription $subscriptionName -AckDeadline -30 -ErrorAction Stop } |
+                Should Throw "Invalid ack deadline given"
+            { New-GcpsSubscription -Topic $topicName -Subscription $subscriptionName -AckDeadline -1000 -ErrorAction Stop } |
+                Should Throw "Invalid ack deadline given"
+        }
+        finally {
+            gcloud beta pubsub topics delete $topicName --quiet 2>$null
+        }
+    }
+
+    It "should error out if topic does not exist" {
+        $r = Get-Random
+        $topicName = "gcloud-powershell-non-existent-topic"
+        $subscriptionName = "gcp-test-new-subscription-$r"
+
+        try {
+            { New-GcpsSubscription -Topic $topicName -Subscription $subscriptionName -ErrorAction Stop } |
+                Should Throw "Topic 'projects/$project/topics/$topicName' does not exist"
+        }
+        finally {
+            gcloud beta pubsub topics delete $topicName --quiet 2>$null
+        }
+    }
+
+    It "should error out if subscription already exists" {
```

Comment:
This doesn't error out at all and looks like a copy of "New-GcpsTopic" "should work"